sktime/experiments/orchestrator.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+import json
+import re
+import numpy as np
+import pandas as pd
+
+
+class Orchestrator:
+    """
+    Orchestrates the sequencing of running the machine learning experiments.
+    """
+    def __init__(self, tasks, datasets, strategies, cv, result):
+        """
+        Parameters
+        ----------
+        tasks: sktime.highlevel.Task
+            task object
+        datasets: pandas dataframe
+            datasets in pandas skitme format
+        strategies: list of sktime strategy
+            strategy as per sktime.highlevel
+        cv: sklearn.model_selection cross validation
+            sklearn cross validation method. Must implement split()
+        result: sktime result class
+            Object for saving the results
+        """
+        self._tasks = tasks
+        self._datasets = datasets
+        self._strategies = strategies
+        self._cv = cv
+        self._result = result
+
+
+    def run(self, predict_on_runtime=True, save_strategies=True):
+        """
+        Method for running the orchestrator
+        
+        Parameters
+        ----------
+        predict_on_runtime:Boolean
+            If True makes predictions after the estimator is trained
+        save_strategies: Boolean
+            If True saves the trained strategies on the disk
+        """
+        
+        for task, data in zip(self._tasks, self._datasets):
+            dts_loaded = data.load()
+            for strategy in self._strategies:
+                for cv_fold, (train, test) in enumerate(self._cv.split(dts_loaded)):
+                    
+                    strategy.fit(task, dts_loaded.iloc[train])
+
+                    if predict_on_runtime:
+                        y_pred = strategy.predict(dts_loaded.iloc[test])
+                        y_true = dts_loaded[task.target].iloc[test].values
+                        self._result.save(dataset_name=data.dataset_name, 
+                                          strategy_name=strategy.name, 
+                                          y_true=y_true.tolist(), 
+                                          y_pred=y_pred.tolist(),
+                                          cv_fold=cv_fold)
+                    if save_strategies:
+                        strategy.save(dataset_name=data.dataset_name, 
```

Comment:
I think the strategy `save` method should not take the dataset and cv as input args, instead the `self._results` object should have a `save_fitted_strategy` class which determines the correct path and then calls the strategy's `save` method.

Reply:
not sure how this will work. You need to provide the path when saving the strategy. What you can do is instead of providing the different arguments that form the path separately you can create the path string inside the results.save_fitted_strategy() method and than call strategy.save(path). However, I don't think that it will make the code more readable.

Reply:
Yes, exactly, that's what I mean